ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java

```diff
@@ -405,11 +405,9 @@ protected void updateOutputs(Table targetTable) {
    */
   private boolean isTargetTable(Entity entity, Table targetTable) {
     //todo: https://issues.apache.org/jira/browse/HIVE-15048
-    /**
-     * is this the right way to compare?  Should it just compare paths?
-     * equals() impl looks heavy weight
-     */
-    return targetTable.equals(entity.getTable());
+    // Since any DDL now advances the write id, we should ignore the write Id,
+    // while comparing two tables
+    return targetTable.equalsWithIgnoreWriteId(entity.getTable());
```

Comment:
Is this the only place where the writeId change will disrupt the use of the table.equals method? I am concerned about this would cause issues with set operations like contains/hashes etc. Do not have a good suggestion. Only concerns yet.

Reply:
I believe this is the only place. Also, everywhere else, we use the default equals method, so I don't see any issues when Table is used within collections. I mean we are not modifying existing equals() method.

Reply:
My concern is mainly about having multiple Table instances in memory around compilation/execution which previously were "equal", but after this change will not be "equal" because the writeId changes. You identified one place where we face this issue, but there might be other places - even possibly hidden ones like collections -, where the objects were previously equals, but not anymore.

Reply:
Ok. Let's shelve this